### PR TITLE
allow import files to have both Zetkin ID and external ID

### LINKED ISSUE #1935: Import: Not possible to import External ID using Zetkin IDs as the key
## Description
A somewhat common pattern will be for users to export a list from Zetkin, populate it with external IDs (e.g. member IDs) and then import that list back. That's currently not possible, because it's not possible to map both Zetkin ID and External ID to (different) columns.

## Steps to reproduce
1. Go to http://localhost:3000/organize/1/people
2. Click on "Create..."
3. Click on "Import people"
4. Import a file that contains at least two columns
5. Map the first column to Zetkin ID
6. Map the second column to External ID

### Expected Behaviour
It should be possible.

### Actual Behaviour
The "ID" option is disabled in the drop-down, because it has already been configured as Zetkin ID.

## Screenshots (if you have any)
N/A

### PR DESCRIPTION
**Pull request by remipou**
## Description
This PR allows to select both zetkinID and externalId when importing files

## Changes
- allow selecting both IDs
- add "member_id" to the person fields (as "ext_id" was previously called "member_id")
- fix a typo

## Related issues
Resolves #1935 (https://github.com/zetkin/app.zetkin.org/issues/1935)

### CODE REVIEW

**Comment by remipou** [comment]
I added "member_id" into some types, it should be removed when the backend replies with "ext_id" instead of "member_id" like it should? @richardolsson

File: src/features/import/hooks/useColumn.ts [hunk 1]
```diff
@@ -42,12 +42,6 @@ export default function useColumn(orgId: number) {
       );
     }
 
-    if (value == 'id') {
-      return !!allSelectedColumns.find(
-        (column) => column.kind == ColumnKind.ID_FIELD
-      );
-    }
-
```
Comment (richardolsson) [inline src/features/import/hooks/useColumn.ts:50]: At first I thought, what a beautifully simple solution. But then I realized, wait, doesn't this mean you can pick multiple "External IDs", and yes you can. It should be possible to pick one Zetkin ID and one External ID, but no more than that. But right now you can:

image

(ignore the fact that I've used completely nonsensical data in my screenshot above)

I don't know how best to solve this. I guess we could merge without making any changes to this, but I would ideally like to at least prevent the user from clicking the "VALIDATE" button if there are more than one Zetkin ID and/or more than one External ID.

Would you mind trying to add logic for that? And then maybe create a separate issue for coming up with an even better solution at some later point in time?

**Review by richardolsson — CHANGES_REQUESTED** [review]
Great work on this! You've managed to solve some things, but I realized as I was playing around with it that it also causes some other problems. I've left some comments below. Do you think you could take a look at them? Please feel free to come talk to me if you have any questions!

File: src/utils/types/zetkin.ts [hunk 2]
```diff
@@ -170,6 +170,7 @@ export interface ZetkinPersonNativeFields {
   street_address: string | null;
   co_address: string | null;
   ext_id: string | null;
+  member_id?: string | null;
```
[the remark this answers is not in the document]
Comment (remipou) [reply src/utils/types/zetkin.ts]: this should solve it? https://github.com/zetkin/app.zetkin.org/pull/2201/commits/272f9c0f00a194ce4fe1a5fefa189de43177e03d

**Comment by remipou** [comment]
ready for review @richardolsson

**Comment by richardolsson** [comment]
Superseded by #2391, where @ziggabyte started from this great work, fixed a merge conflict and made some minor changes to the UX.

**Pull request closed without merging**